editor/gui/scene_tree_editor.cpp

```diff
@@ -1501,11 +1527,33 @@ void SceneTreeEditor::rename_node(Node *p_node, const String &p_name, TreeItem *
 			error->popup_centered();
 		}
 		item->set_text(0, p_node->get_name());
+		if (p_node->is_unique_name_in_owner()) {
+			return;
+		}
+	}
+
+	// If same name and check_for_percent_token is still true, now set as unique.
+	// This is separate from final action so "Rename Node" is not added to undo history.
```

Comment:
Wouldn't it be better if it was part of the same action? Otherwise when you rename node with `%` you get 2 undo actions.